sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/DefaultComponentTree.java

```diff
@@ -41,13 +38,8 @@ public void index(InputComponent component, InputComponent parent) {
     Preconditions.checkNotNull(component);
     Preconditions.checkNotNull(parent);
     parents.put(component, parent);
-    Set<InputComponent> list = children.get(parent);
-    if (list == null) {
-      list = new LinkedHashSet<>();
-      children.put(parent, list);
-    }
-
-    list.add(component);
+    children.putIfAbsent(parent, new LinkedHashSet<>());
+    children.get(parent).add(component);
```

Comment:
Sorry I've read too quickly. I though you proposed to use `children.putIfAbsent(parent, new LinkedHashSet<>()).add(component);`